packages/scene-composer/src/components/panels/CustomColorPicker/CustomColorPicker.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useRef, useState } from 'react';
+import { HexColorPicker } from 'react-colorful';
+import { debounce } from 'lodash';
+import { Button, FormField, InputProps, SpaceBetween } from '@awsui/components-react';
+import './CustomColorPicker.scss';
+import { useIntl } from 'react-intl';
+
+import {
+  ColorPickerDivider,
+  ColorPickerWrapper,
+  ColorPreview,
+  FlexibleDiv,
+  RGB,
+  RgbToHex,
+  hexToRgb,
+  isValidHex,
+  validateHex,
+  validateRgbValue,
+  FixedWidthInput,
+} from './CustomColorPickerHelpers';
+
+export const DEFAULT_COLOR_PICKER_COLOR = {
+  hex: '#ffffff',
+  rgb: {
+    r: '255',
+    g: '255',
+    b: '255',
+  },
+};
+
+interface CustomColorPickerProps {
+  color: string | undefined;
+  onSubmit: (newColor: string) => void;
+  onCancel?: () => void;
+}
+
+export const CustomColorPicker: React.FC<CustomColorPickerProps> = ({
+  color,
+  onSubmit,
+  onCancel,
+}: CustomColorPickerProps) => {
+  const { formatMessage } = useIntl();
+  const hexInputFieldRef = useRef<InputProps.Ref>(null);
+  const [saveAttempted, setSaveAttempted] = useState(false);
```

Comment:
I still feel we should make this save optional as it makes easier for customer to just select a color from the picker. We need to check with @brenton

Reply:
Sounds good, looks like he should be back tomorrow. I'll ping him on slack and see what he thinks, and I'll include you in the chat